File: 2033_minimum_operations_to_make_a_uni_value_grid.py
class Solution(object):
    def minOperations(self, M, d):
        a = sum(M, [])
        b = a[0]
        if any((i - b) % d for i in a): return -1
        a.sort()
        m = a[len(a)//2]
        return sum(abs(i - m) // d for i in a)

File: 2033_minimum_operations_to_make_a_uni_value_grid.java
class Solution {
    public int minOperations(int[][] m, int d) {
        List<Integer> a = new ArrayList<>();
        for (int[] r : m) for (int n : r) a.add(n);
        int b = a.get(0);
        for (int n : a) if ((n - b) % d != 0) return -1;
        Collections.sort(a);
        int t = a.get(a.size()/2), s = 0;
        for (int n : a) s += Math.abs(n - t) / d;
        return s;
    }
}
